pac4j-core/src/main/java/org/pac4j/core/io/Resource.java

```diff
@@ -0,0 +1,32 @@
+package org.pac4j.core.io;
+/**
+ * Interface for a resource descriptor that abstracts from the actual
+ * type of underlying resource, such as a file or class path resource.
+ *
+ * <p>An InputStream can be opened for every resource if it exists in
+ * physical form, but a URL or File handle can just be returned for
+ * certain resources. The actual behavior is implementation-specific.
+ *
+ */
+public interface Resource extends InputStreamSource {

```

Comment:
But no other interface extends from `InputStreamSource` directly, doesn't it? So in our case, a `Resource` is necessarily a `InputStreamSource` isn't it? We never distinguish between both interfaces.
Or do you directly use the `InputStreamResource` for your Spring extension?
That said, it might be cleaner to keep both for clarity.